### PR TITLE
fix: fixed a bug where users can enter values while the help modal is present

### LINKED ISSUE #81: User can interact with the underlying window when the modal window is open
When the help dialogue is open, you can type or tab through components on the screen. In Liedle, you can type a word and play the game without closing the dialogue. This is a problem because users will want to read and understand the game before using up their guesses. 

**Steps to reproduce**
- Run `npm start`.
- Visit `http://localhost:1234/liedle/index.html`.
- There should be a dialogue already open. This dialogue can also be brought back with the ❔ symbol in the navigation bar. 
- Type some letters
- Close the window using the x button or by clicking outside of the dialogue
- You will see the letters you just typed

**Potential cause of the bug**
No extra logic to disable the underlying contents has been implemented.

### PR DESCRIPTION
**Pull request by JosiahBull**
## Todo
- [x] add test to validate failure condition
- [x] fix bug by adding event emits from the help modal when opened/closed 

## Motivation

Users should not be able to enter text into the game while the help modal is present. They could accidentally use up guesses, and it would be frustrating.

## Summary of changes

- Help modal in `components/script/help-modal.js` now emits `modal-open` and `modal-close` events at the `document` scope.
- Tests added to validate that text may only be entered while the modal is closed.
- `liedle/index.js` keystroke input respects whether or not the modal is open to allow the user to enter keys.

## Attached GitHub issue

Closes #81

### Local Build
- [x] Ran all local tests `npm run test:ci`
- [x] Manually tested my solution

### GitHub
- [x] Target branch has been set correctly
- [x] PR has been rebased onto target branch
- [x] PR has been assigned an owner
- [x] Author has performed a self-review of the code
- [x] Removed the WIP message from the top of this PR

### CODE REVIEW

**Comment by JosiahBull** [comment]
rebased to fix a merge conflict generated by #113

File: src/liedle/index.js [hunk 1]
```diff
@@ -19,7 +19,9 @@ function start() {
   isEndGame = false;
   createGrid();
   resetGame();
-  handleInput();
```
Comment (lche262) [inline src/liedle/index.js:22]: can you briefly explain why don't we need handleInput() here?

**Review by lche262 — APPROVED** [review]
Nice test case writing! Tested and it is working perfectly

File: src/liedle/index.js (hunk 1 above)
Comment (JosiahBull) [reply src/liedle/index.js:22]: `handleInput()` starts handlers which parse and respond to keypresses, we only want it to be run if we get a `modal-closed` event as by default the modal is open.

Comment (JosiahBull) [reply src/liedle/index.js:22]: `handleInput()` starts handlers which parse and respond to keypresses, we only want it to be run if we get a `modal-closed` event as by default the modal is open and we don't want to respond to keypresses.

**Pull request merged**